src/integrator_interface.jl

```diff
@@ -406,6 +413,15 @@ function get_sol(integrator::DEIntegrator)
     return integrator.sol
 end
 
+"""
+    get_retcode(integrator::DEIntegrator)
+
+Get the return code of the integrator.
+"""
+function get_retcode(integrator::DEIntegrator)
+    return integrator.sol.retcode
+end
```

Comment:
To quickly elaborate on this, I essentially have problems where "time sub-integrators" advance parts of the solution and I could not find a sane way to define this distributed solution stuff into the current solution interface. Technically I do not even need every subintegrator to hold a solution. They just need to communicate return codes between each other. Having this interface here would be a compromise.

Reply:
Most integrators don't set the return code until the end though?

Reply:
Correct. But for operator splitting methods I have essentially a tree of integrators, each solving subproblems on small time intervals. The easiest example is essentially to have an additivley split right hand side for some ODE: 

$d_t u = f(u,p,t) = f_1(u,p,t) + f_2(u,p,t)$

which I want to integrate from [0,T]. Operator splitting method now definine two ODEs

$d_t u^* = f_1(u^*,p,t)$

and

$d_t u^{**} = f_2(u^{**},p,t)$

and they intrgrate each of these in a specific order on subintervals on $[0,T]$. The simplest rule is to solve them alternatingly on intervals with fixed dt. E.g. the first step would be

1. Set $u^*(0) = u_0$
2. Solve $d_t u^* = f_1(u^*,p,t)$ on $[0,dt]$ (with your algorithm of choice)
3. Set $u^{**}(0) = u^*(dt)$
4. Solve $d_t u^{**} = f_2(u^{**},p,t)$ on $[0,dt]$ (with your algorithm of choice)


In my implementation right now I have three separate integrators. One custom time integrator for the coordination of which subintegrator needs to integrate its associated problem next (+ time step controll of the time intervals to solve the subproblems on), and another two integrators for the respective subproblems. To check whether a solve worked or not the outer time integrator essentially checks the inner integrators return code after each solve of the subproblems. Does this explain it?

Probably not the best architecture, but at least it is functional.